packages/server/src/user/user.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { LoginDto } from './dto/login.dto';
+import { RegisterDto } from './dto/register.dto';
+import { User } from './user.interface';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel('User') private readonly userModel: Model<User>) {}
+
+  findAll() {
+    return this.getAll();
+  }
+
+  find(id: string) {
+    return this.get(id);
+  }
+
+  async login(loginDto: LoginDto) {
+    const result = await this.isUserExist(loginDto.email, loginDto.password);
+    return result;
+  }
+
+  register(registerDto: RegisterDto) {
+    const user: User = {
+      name: registerDto.name,
+      email: registerDto.email,
+      password: registerDto.password,
+    };
+    return this.addUser(user);
+  }
+
+  private async addUser(user: User) {
+    const newUser = new this.userModel(user);
+    const result = await newUser.save();
+    return result;
+  }
+
+  private async getAll() {
+    const result = await this.userModel.find({});
+    const updatedResult = result.map((user) => {
```

Comment:
why do we need `map` here can you find a way to tell mongoose return only fields that you need.